src/DBFTPlugin/Consensus/ConsensusService.cs

```diff
@@ -155,6 +155,12 @@ private void OnStart()
                     CheckPreparations();
                     return;
                 }
+
+                if (context.ViewNumber != 0)
```

Comment:
If we are going to proceed, let's just make sure that `ViewNumber` could not assume something different here and lead to a crash.
I did not yet check it completely.

Reply:
I dont think this is a good idea to assume a CN can resume from the most latest context. Looks like this patch only work when the node restarts the consenssu right after it crashes. 
BTW, if a node crashes, it is indeed a byzantine node.

Reply:
Byzantine is different than a faulty.
This is a Bug. An honest node wil behave byzantine.
When we designed dbft 2.0 this was missed.

Reply:
@vncoelho I understand your concern, what i mean is that the previous context might not work if you restart the node longer than 15 seconds after the node failure.

Reply:
Yes, if a new block is generated it is discarted.

If view is again changed, time is exponential, there is more chance that it will work.

Reply:
Do we have something like a `deamon` to bring the node back to live right after it fails?

Reply:
This depends on each node provider. Basically it is just open cli again.
We should provide them with the correct cache capabilities.

Reply:
> What you added here is more like a mitigation instead of a solution.

I'd say an improvement for some subset of situations. Because a node restart is a byzantine event, in general the node does not follow the protocol in this case, so the outcome can be anything. But this save/restore logic allows to recover in some cases, true. In other cases you might have a disk failure, a DB issue, anything that will prevent the node from recovering and it'll have to start over from nothing. So it's kinda OK to not restore as long as we have no more than f such nodes.

Reply:
Your are misunderstanding a Byzantine event and a software bug.

Reply:
It is not OK to not restore.
There is no reason to let the node reinitialize wrong.

This is not a behavior we expected initially, I also belive we had this working properly in past versions but we missed in some final PR adjustments.

Reply:
> If we are going to proceed, let's just make sure that `ViewNumber` could not assume something different here and lead to a crash. I did not yet check it completely.

My comment from Dec,2022